sources/packages/backend/libs/integrations/src/institution-integration/ece-integration/ece-response.processing.service.ts

```diff
@@ -74,41 +67,54 @@ export class ECEResponseProcessingService {
    * @returns Process summary result.
    */
   async process(): Promise<ProcessSummaryResult[]> {
-    // Get all the institution codes who are enabled for integration.
-    const integrationEnabledInstitutions =
-      await this.institutionLocationService.getAllIntegrationEnabledInstitutionCodes();
-    const filePathDetails: InstitutionFileDetail[] =
-      integrationEnabledInstitutions.map((institutionCode) => ({
-        path: path.join(
-          this.institutionIntegrationConfig.ftpResponseFolder,
-          institutionCode,
-          ECE_RESPONSE_FILE_NAME,
-        ),
-        institutionCode,
-      }));
+    // Get all the ECE response files from the SFTP location.
+    const filePaths = await this.integrationService.getResponseFilesFullPath(
+      this.institutionIntegrationConfig.ftpResponseFolder,
+      /^CONR-008-[A-Z]{4}-\d{8}-\d{6}\.TXT$/i,
+    );
+
+    if (!filePaths.length) {
+      this.logger.log("No ECE response files found to be processed.");
+      return [];
+    }
+    this.logger.log(
+      `Received ${filePaths.length} ece response file(s) to process.`,
+    );
+
     // Process all the files in parallel.
-    const result = await processInParallel<
-      ProcessSummaryResult,
-      InstitutionFileDetail
-    >(
-      (remoteFileDetail: InstitutionFileDetail) =>
-        this.processDisbursementsInECEResponseFile(remoteFileDetail),
-      filePathDetails,
+    const result = await processInParallel<ProcessSummaryResult, string>(
+      (remoteFilePath: string) =>
+        this.processDisbursementsInECEResponseFile(remoteFilePath),
+      filePaths,
     );
     return result;
   }
 
   /**
    * Process individual ECE response file sent by institution.
-   * @param institutionFileDetail file details.
+   * @param remoteFilePath remote file path.
    * @returns process summary result.
    */
   private async processDisbursementsInECEResponseFile(
-    institutionFileDetail: InstitutionFileDetail,
+    remoteFilePath: string,
   ): Promise<ProcessSummaryResult> {
-    const institutionCode = institutionFileDetail.institutionCode;
-    const remoteFilePath = institutionFileDetail.path;
+    const fileName = path.basename(remoteFilePath);
     const processSummary = new ProcessSummaryResult();
+    // The file name is expected to match the pattern 'CONR-008-XXXX-....'.
+    // The institution code is the 4 alpha characters after 'CONR-008-'.
+    const institutionCode = fileName.substring(9, 13);
+    const integrationLocation =
+      await this.institutionLocationService.getIntegrationLocation(
+        institutionCode,
+      );
+
+    if (!integrationLocation) {
+      processSummary.summary.push(
```

Comment:
In this case, it must be pushed to processSummary.warnings IMO as we need to be aware of a file that came to us with institution code either not existing in our system or integration flag not enabled.